crates/solidity/src/zksolc/main.rs

```diff
@@ -190,18 +194,36 @@ fn main_inner() -> anyhow::Result<()> {
         );
     } else if arguments.output_assembly || arguments.output_binary {
         for (path, contract) in build.contracts.into_iter() {
+            let bytescode = contract.build.bytecode;
+
             if arguments.output_assembly {
-                println!(
-                    "Contract `{}` assembly:\n\n{}",
-                    path, contract.build.assembly_text
-                );
+                let program_blob = match ProgramBlob::parse(bytescode.as_slice()) {
+                    Ok(blob) => blob,
+                    Err(error) => {
+                        bail!("Failed to parse program blob: {}", error);
+                    }
+                };
+
+                let disassembler_object =
+                    Disassembler::new(&program_blob, DisassemblyFormat::Guest).context(format!(
+                        "Failed to create disassembler for contract '{}'",
+                        path
+                    ))?;
+
```

Comment:
I think we could instruct the disassembler to emit gas usage (`disassembler.display_gas()?;`).

Reply:
yep, I will add it now